### PR TITLE
[topi][relay] Add operation scatter_add to relay, based on scatter implementation.

### PR DESCRIPTION
**Pull request by notoraptor**
Hi! This is a pull request to add operation `scatter_add` in relay. Code is based on implementation of `scatter` previously added by @mbrookhart .

### CODE REVIEW

File: tests/python/relay/test_op_level3.py [hunk 1]
```diff
@@ -811,6 +811,51 @@ def verify_scatter(dshape, ishape, axis=0):
...
+        z = relay.op.scatter_add(d, i, u, axis)
+
+        func = relay.Function([d, i, u], z)
+
+        data_np = np.random.uniform(size=dshape).astype("float32")
+        updates_np = np.random.uniform(size=ishape).astype("float32")
+        indices_np = np.random.randint(-dshape[axis], dshape[axis] - 1, ishape).astype("int64")
+
+        ref_res = ref_scatter_add(data_np, indices_np, updates_np, axis)
+        # TODO(mbrookhart): expand testing when adding more backend schedules
```
Comment (mbrookhart) [inline tests/python/relay/test_op_level3.py:837]: Any chance you got it to work on cuda? I haven't gotten back to this TODO yet :)

Comment (notoraptor) [reply tests/python/relay/test_op_level3.py:837]: Hi! Unfortunately, no! For the moment, I just need it on CPU ^^ !

**Comment by notoraptor** [comment]
I got an unexpected error in jenkins:

```
running 3 tests

test graph::tests::test_str_to_type ... ok

test threading::tests::test_max_concurrency ... ok

test threading::tests::test_parallel_launch ... FAILED


failures:


---- threading::tests::test_parallel_launch stdout ----

thread 'threading::tests::test_parallel_launch' panicked at 'assertion failed: `(left == right)`

  left: `1`,

 right: `3`', runtime/src/threading.rs:244:13

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It seems to come from Rust tests. I don't know why.

I just rebased the branch to relaunch tests again.

**Comment by notoraptor** [comment]
All tests passed, this time !

**Comment by mbrookhart** [comment]
@Laurawly @Huyuwei Could you take a look?

**Comment by tqchen** [comment]
Thanks @notoraptor , @mbrookhart !

**Pull request merged**